tracer/transport.go

```diff
@@ -22,6 +25,24 @@ type Transport interface {
 	SetHeader(key, value string)
 }
 
+// NewHTTPTransport returns a new Transport implementation that sends traces to
+// a trace agent running on the given hostname and port. If the zero values for
+// hostname and port are provided, the default values will be used ("localhost"
+// for hostname, and "7777" for port).
+//
+// In general, using this method is only necessary if you have a trace agent
+// running on a non-default port or if it's located on another machine.
+func NewHTTPTransport(hostname, port string) Transport {
+	if hostname == "" {
+		hostname = defaultHostname
+	}
+	if port == "" {
+		port = defaultPort
+	}
+	url := fmt.Sprintf("http://%s:%s/v0.3/traces", hostname, port)
```

Comment:
Oh, sorry but I spotted that problem while addressing your PR. Mainly, if for any reason we need to downgrade the API to a stable version, we lose your changes since [we're using][1] a ``legacyDeliveryURL``. To be consistent, we should handle both cases otherwise in the future we forget about that, having an unexpected behavior.

Anyway in the overall it looks fine! Just this change and we're ready to go.

[1]: https://github.com/ejholmes/dd-trace-go/blob/44c9be8c0200a0ba12fa1a2849665b3768fe2ec5/tracer/transport.go#L132-L136